module/modmanager/module.go

```diff
@@ -123,10 +122,45 @@ func (m *module) checkReady(ctx context.Context, parentAddr string) error {
 	}
 
 	for {
+		var resp *pb.ReadyResponse
 		// 5000 is an arbitrarily high number of attempts (context timeout should hit long before)
-		resp, err := m.client.Ready(ctxTimeout, req, grpc_retry.WithMax(5000))
-		if err != nil {
-			return err
+		for range 5000 {
+			perCallCtx, perCallCtxCancelFunc := context.WithTimeout(parentCtxTimeout, 3*time.Second)
+			resp, err = m.client.Ready(perCallCtx, req)
+			perCallCtxCancelFunc()
+
+			// if module is not ready yet, wait and try again
+			code := status.Code(err)
+			// context errors here are the perCallCtx
+			if code == codes.Unavailable || code == codes.ResourceExhausted || code == codes.DeadlineExceeded || code == codes.Canceled {
+				waitTimer := time.NewTimer(200 * time.Millisecond)
+				select {
+				case <-parentCtxTimeout.Done():
+					waitTimer.Stop()
+					return parentCtxTimeout.Err()
+				case <-waitTimer.C:
+				}
+				// Short circuit this check if the process has already exited. We could get here if process exits before
+				// module can set up the Ready server.
+				// This is most likely in TCP mode, where the .sock presence check is skipped, but
+				// also possible in UNIX mode if the process exits in between.
+				// (OUE is waiting on the same modmanager lock, so it can't try to restart).
+				if errors.Is(m.process.Status(), os.ErrProcessDone) {
+					m.logger.Info("Module process exited unexpectedly while waiting for ready.")
+					parentCtxCancelFunc()
+					return parentCtxTimeout.Err()
```

Comment:
shouldn't we return the info log as an error instead of a context canceled error? seems like a bit of an obfuscation

Reply:
I don't mean to log `m.logger.Info("Module process exited unexpectedly while waiting for ready.")` as an error level log, more that the returned error should be `"module process exited unexpectedly while waiting for ready."` IMO or `"module process exited unexpectedly` so that the full message would be `error while waiting for module to be ready viam_rsdk12369-3: module process exited unexpectedly`